File: Python3/solutions/two_sum.py
from typing import List

class Solution1:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        seen = {}
        for i in range(len(nums)):
            value = target-nums[i]
            if value in seen:
                return [seen[value],i]
            seen[nums[i]] = i

class Solution2:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(len(nums)):
            for j in range(i+1, len(nums)):
                if nums[i] + nums[j] == target:
                    return [i,j]

File: C++/solutions/two_sum.cpp
// 1. Two Sum
// Problem: https://leetcode.com/problems/two-sum/

// Runtime: 10 ms
// Memory Usage: 13.69 MB
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> mymap;
        mymap[nums[0]] = 0;
        for(int i=1;i<nums.size();i++){
            if (mymap.find(target-nums[i]) != mymap.end()) {
                return {mymap[target-nums[i]], i};
            } else {
                mymap[nums[i]] = i;   
            }
        }
        return {};
    }
};
